tests/perf.c

```diff
@@ -605,70 +873,260 @@ sysrepo_init(void)
+    {"create an item cached", setup_running_cached, test_item_create, teardown_running},
+    {"create an item oper", setup_oper, test_item_create_oper, teardown_oper},
+    {"modify an item", setup_running, test_item_modify, teardown_running},
+    {"modify an item cached", setup_running_cached, test_item_modify, teardown_running},
+    {"remove an item", setup_running, test_item_remove, teardown_running},
+    {"remove an item cached", setup_running_cached, test_item_remove, teardown_running},
+};
+
+void
+print_top_table_boundary(const char *plugin_name)
+{
+    uint32_t i;
+    const char *first = " test name ", *second = " time ";
+    const char *third = " comparison ";
+
+    printf("\n  %s\n", plugin_name);
+
+    printf(" ");
+    for (i = 0; i < TABLE_WIDTH; ++i) {
+        printf("_");
+    }
+    printf(" \n|");
+    for (i = 0; i < TABLE_WIDTH; ++i) {
+        printf(" ");
+    }
+
+    printf("|\n|");
+    printf("%s", first);
+    for (i = strlen(first); i <= NAME_FIXED_LEN + 2; ++i) {
+        printf(" ");
+    }
+    printf("|");
+    for (i = strlen(second); i < COL_FIXED_LEN - 1; ++i) {
+        printf(" ");
+    }
+    printf("%s", second);
+    printf("|");
+    for (i = strlen(third); i < COL_FIXED_LEN; ++i) {
+        printf(" ");
+    }
+    printf("%s", third);
+    printf("|\n");
+
+    // top table boundary
+    printf("|");
+    for (i = 0; i < TABLE_WIDTH; ++i) {
+        printf("_");
+    }
+    printf("|\n");
+
+    printf("|");
+    for (i = 0; i < TABLE_WIDTH; ++i) {
+        printf(" ");
+    }
+
+    printf("|\n");
+}
+
+void
+print_test_results(const char *test_name, int64_t time, int64_t dflt_time)
+{
+    uint32_t printed;
+    long double ratio = 1.0f;
+    uint64_t ratio_decimal;
+
+    printf("| %s ", test_name);
+    printed = strlen(test_name);
+    while (printed < NAME_FIXED_LEN) {
+        printf(".");
+        printed += 1;
+    }
+    printf(" | %3" PRId64 ".%06" PRId64 " s |", time / MILLION, time % MILLION);
+    printf(" ");
+
+    if (time < dflt_time) {
+        printf("\033[0;32;1m");
+        ratio = ABS((long double)(dflt_time) / (long double)(time));
+    } else if (time > dflt_time) {
+        printf("\033[0;31;1m");
+        ratio = ABS((long double)(time) / (long double)(dflt_time));
+    }
+    ratio_decimal = (uint64_t)((ratio - (uint64_t)(ratio)) * 1000);
+
+    printf("%7" PRIu64 ".%03" PRIu64 " x ", (uint64_t)(ratio), ratio_decimal);
+    printf("\033[0;37;1m");
+    printf("|\n");
+}
+
+void
+print_bottom_table_boundary()
+{
+    uint32_t i;
+
+    printf("|");
+    for (i = 0; i < TABLE_WIDTH; ++i) {
+        printf("_");
+    }
+    printf("|\n\n");
+}
+
 int
 main(int argc, char **argv)
 {
-    int ret;
-    uint32_t i, count, tries;
+    int ret = 0;
+    uint32_t i, j, count, tries, plg_cnt, test_cnt;
+    const char *plg_name;
+    int64_t *times = NULL, time;
+    struct test_state state = {0};
 
+    /* change print color */
+    printf("\033[0;37;1m");
```

Comment:
I would be wary of such commands, have you checked compatibility of various shells with this?

Reply:
All modern terminal emulators use ANSI escape codes